domain/src/main/kotlin/feature/BackupManager.kt

```diff
@@ -0,0 +1,58 @@
+package ltd.evilcorp.domain.feature
+
+import android.content.res.Resources
+import android.net.Uri
+import androidx.core.os.ConfigurationCompat
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ltd.evilcorp.core.repository.MessageRepository
+import ltd.evilcorp.core.vo.Message
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.collections.List
+
+@Singleton
+class BackupManager @Inject constructor(
+    private val messageRepository: MessageRepository,
+) {
+
+    companion object {
+        private val mapper = jacksonObjectMapper()
+    }
+
+    private fun List<Message>.generateBackupMessages(locationSave: String): BackupMessages {
```

Comment:
Make these normal functions instead of an extension of List<Message> to make it clearer what's going on.

Reply:
Okay, though it would make it less streamy, then.

https://github.com/evilcorpltd/aTox/blob/d39dc24119a011b1cdbec9a98870b5c0586adc01/domain/src/main/kotlin/feature/BackupManager.kt#L44-L51